openhcl/virt_mshv_vtl/src/processor/hardware_cvm/mod.rs

```diff
@@ -635,7 +640,33 @@ impl<B: HardwareIsolatedBacking> UhProcessor<'_, B> {
                 }
             }
             GuestVtl::Vtl1 => {
-                // TODO: Check for VINA
+                // Check for VINA
+                if let Some(vector) = interrupt_pending[GuestVtl::Vtl0] {
+                    let priority = vector >> 4;
+                    if priority > 0 {
+                        let hv = self.hv[GuestVtl::Vtl1].as_mut().unwrap();
+                        if hv.synic.vina().enabled() && !hv.vina_asserted().unwrap() {
+                            let ppr = lapic_msr_read(
+                                self,
+                                GuestVtl::Vtl1,
+                                ApicRegister::PPR.0 as u32 + x86defs::apic::X2APIC_MSR_BASE,
+                            )
+                            .expect("reading PPR should never fail");
+                            let ppr_priority = ppr >> 4;
+                            if priority as u64 > ppr_priority {
+                                let vp_index = self.vp_index();
+                                let hv = self.hv[GuestVtl::Vtl1].as_mut().unwrap();
+                                hv.set_vina_asserted(true).unwrap();
+                                self.partition
+                                    .synic_interrupt(vp_index, GuestVtl::Vtl1)
+                                    .request_interrupt(
```

Comment:
Does the interrupt still get delivered, if we're doing this after the apic poll?

Reply:
Good catch. I'm not sure if this is enough or if we need to poll the apic too here.

Reply:
I think that's probably a sign that we're doing this in the wrong order. We probably want to evaluate VINA before we poll the VTL1 APIC, to avoid redundant work. But some of this depends on my other line of questioning.

Reply:
I think the answer is we want to do it inside the loop, and then loop again if vina gets asserted. That'd reevaluate everything, right?